cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -222,13 +180,25 @@ func run(_ <-chan struct{}) {
 
 	scaleDown := NewScaleDown(&autoscalingContext)
 
+	configFetcher := dynamic.NewConfigFetcher(*configMapName, KubernetesNamespace, kubeClient)
+	dynamicReconfiguration := NewDynamicReconfiguration(*configMapName, &autoscalingContext, configFetcher, cloudProviderBuilder)
+
 	for {
 		select {
 		case <-time.After(*scanInterval):
 			{
 				loopStart := time.Now()
 				updateLastTime("main")
 
+				if dynamicReconfiguration.Enabled() {
+					reconfigureStart := time.Now()
+					updateLastTime("reconfigure")
+					if err := dynamicReconfiguration.Run(); err != nil {
+						glog.Errorf("Failed to reconfigure : %v", err)
+					}
+					updateDuration("reconfigure", reconfigureStart)
+				}
```

Comment:
For safety, any config change should stop and recreate all the stuff running in CA. I imagine that we would like to change not only the node groups but also thresholds and other settings.